server/api/handlers/game_action.go

```diff
@@ -31,6 +31,10 @@ func ProcessGameActionRequest(context context.Context, req *ws_interface.ApiRequ
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
+	if session.State == models.GameFailed {
```

Comment:
what if closed(Finished) ?

Reply:
if closed then state should be failed, shouldn't it? https://github.com/DaoCasino/game-contract-sdk/blob/master/sdk/include/game-contract-sdk/game_base.hpp#L526

Reply:
does thi PR solve the problem?
@Satorit wanted 'session expired' to be displayed, looks like we won't
be able to do that if we merge it

Reply:
I checked `game_base.hpp` looks like `game_failed` is only emitted in `close` action. 
Given how unlikely we are to rename it in the contract in my opinion we can assume that `game_failed = session expired`

Reply:
need to rename error to `SessionFailedOrFinished` and return it if `session.State == models.GameFailed || session.State == models.GameFinished`

Reply:
done